### DISCUSSION TITLE
Clarification of strata argument for adoinis2

### DISCUSSION THREAD

**Discussion by cazzlewazzle89** [vegandevs/vegan#604] [Q&A]
Hi all,

As the title suggests, I'm hoping for some clarification on the `strata` argument in `adonis2`.
I have paired samples from the faecal microbiome of mice, one is a 'total' faecal sample, the other is an extract of bacterial extracellular vesicles from the same sample. 
I also have two treatment groups.

I am trying to see whether Treatment impacts the extracellular vesicle-producing component of the microbiome, which using the paired 'total' faecal sample to control for the fact that the microbiome will differ between individual mice. 
Should I use MouseID as the `strata` argument for PERMANOVA?

A standard two way PERMANOVA gives me no significant effect of Treatment
```
adonis2(formula = bray_dist_paired ~ metadata_paired$Source * metadata_paired$Treatment)
                                                 Df SumOfSqs      R2       F Pr(>F)    
metadata_paired$Source                            1   4.7516 0.43150 26.7180  0.001 ***
metadata_paired$Treatment                         1   0.2809 0.02551  1.5795  0.142    
metadata_paired$Source:metadata_paired$Treatment  1   0.2882 0.02618  1.6208  0.134    
Residual                                         32   5.6910 0.51681                   
Total                                            35  11.0117 1.00000                   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ‘ ’ 1
```
But adding the strata argument makes it significant
```
adonis2(formula = bray_dist_paired ~ metadata_paired$Source * metadata_paired$Treatment, strata = metadata_paired$MouseID)
                                                 Df SumOfSqs      R2       F Pr(>F)    
metadata_paired$Source                            1   4.7516 0.43150 26.7180  0.001 ***
metadata_paired$Treatment                         1   0.2809 0.02551  1.5795  0.001 ***
metadata_paired$Source:metadata_paired$Treatment  1   0.2882 0.02618  1.6208  0.275    
Residual                                         32   5.6910 0.51681                   
Total                                            35  11.0117 1.00000                   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ‘ ’ 1
```

I am unsure which method is correct. The R2 value doesn't change, so I think I'm just unsure of what `strata` actually does. 

Thanks in advance for any advice,
Calum

**Comment by gavinsimpson** [comment]
As per `?adonis2`:

> `strata` 
> Groups within which to constrain permutations. The traditional non-movable strata are set as Blocks in the [permute](https://cran.r-project.org/package=permute) package, but some more flexible alternatives may be more appropriate.

Hence `strata` doesn't change the model (which is why the estimated effects don't change), but it does affect the permutations. As per the documentation, permutations are restricted to be *within* the levels of `strata`. In your case this means we shuffle at random each pair of samples from an individual mouse. As I doubt (as your description is not complete &mdash; "I have two treatment groups" is, almost entirely, uninformative) the treatment varies at the `Source` level (i.e. in your setup, each mouse only experiences one of the two treatment levels), it seems to me that neither permutation test is correct.

If I surmise correctly, you want to permute the groups of paired samples, while keeping the samples within the pair unpermuted.

I also don't understand what you were hoping to do with the `Source` variable &mdash; all that can be explaining is the differences in "average" composition of the samples belonging to each `Source`, which doesn't sound like it will "control for the fact that the microbiome will differ between individual mice". If you want to to do that, you would need to include `MouseID` as a parametric factor term in the model effectively as a nuisance term.

You're going to have to explain more (generically - I don't want to know about the scientific details of your study, just the experimental design elements) if you want more specific help. Absent that, all I can suggest right now is that neither of your permutation designs is correct and your model is incorrectly specified for the effects you wish to estimate.

**Comment by jarioksa** [comment]
Does the Discussion #600 help you? Have you read the vignette of the **permute** package?

If your problem is "paired", you cannot use `strata`, but you must define your permutation scheme in the **permute** package. See help in the **permute** package.
